src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -145,11 +162,15 @@ private void OnOptionChanged(object sender, OptionChangedEventArgs e)
                     SolutionCrawlerLogger.LogOptionChanged(CorrelationId, value);
                     return;
                 }
+                else if (e.Option == SolutionCrawlerOptions.BackgroundAnalysisScopeOption)
+                {
+                    forceReanalysis = true;
+                }
 
-                ReanalyzeOnOptionChange(sender, e);
+                ReanalyzeOnOptionChange(sender, e, forceReanalysis);
             }
 
-            private void ReanalyzeOnOptionChange(object sender, OptionChangedEventArgs e)
+            private void ReanalyzeOnOptionChange(object sender, OptionChangedEventArgs e, bool forceReanalysis)
```

Comment:
actually, you pass in 'e' here as well.  why can't this value be inferred inside this method instead of needing to be passed in?